clang/test/C/C2y/n3353.c

```diff
@@ -0,0 +1,117 @@
+// RUN: %clang_cc1 -verify=expected,c2y -pedantic -std=c2y %s
+// RUN: %clang_cc1 -verify=expected,c2y,compat -Wpre-c2y-compat -std=c2y %s
+// RUN: %clang_cc1 -verify=expected,ext -pedantic -std=c23 %s
+// RUN: %clang_cc1 -verify=expected,cpp -pedantic -x c++ -Wno-c11-extensions %s
+
+
+/* WG14 N3353: Clang 21
+ * Obsolete implicitly octal literals and add delimited escape sequences
+ */
+
+constexpr int i = 0234;  // c2y-warning {{octal literals without a '0o' prefix are deprecated}}
```

Comment:
Maybe it would be nice to also show octal literals in array indices, math etc to be more complete in all the place we could find them.

Reply:
Each of those cases is as an expression, and we have coverage for expressions already (e.g., `0O1234 == 0o1234`), so I'm not certain what other test coverage we'd get from that.

Reply:
Also what about suffixes like `ul` etc and they should be accepted as null pointer constants?

Reply:
yup, valid anywhere you used to be able to write an octal literal before with the same semantics. I'll add some additional tests.